types/project.go

```diff
@@ -572,7 +618,15 @@ func (p Project) ResolveServicesEnvironment(discardEnvFiles bool) error {
 		if discardEnvFiles {
 			service.EnvFiles = nil
 		}
-		p.Services[i] = service
+		newProject.Services[i] = service
 	}
-	return nil
+	return newProject, nil
+}
+
+func (p *Project) deepCopy() (*Project, error) {
+	instance, err := copystructure.Copy(p)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
In theory, this should not fail, right? It might make sense for this to `panic()` to avoid carrying the `error` everywhere.